### PR TITLE
As 979 documentation of configuration options

### PR DESCRIPTION
**Pull request by thinh-askui**

### CODE REVIEW

File: packages/askui-nodejs/src/lib/control-ui-server-args.ts [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
   readonly minimize?: boolean;
+  /* Download the provided Version of askui UI Controller.
+  If a version is already downloaded. This version will be overwritten. */
   readonly overWriteBinary?: boolean;
+  /* You can use different types of Log Level. Options:
+  options: "fatal", "error", "warn", "info", "debug", "trace", "silent", "verbose".
+  */
   readonly logLevel?: LogLevels;
+  /* It is possible to specify a path for your log files.
+  Logs are written to the stdout per default. */
```
Comment (mlikasam-askui) [inline packages/askui-nodejs/src/lib/control-ui-server-args.ts:24]: if you start the binary through the lib, the server logs won't be displayed in the stdout

File: docs/docs/api/06-Configuration/askui-ui-controller-client.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+`DISABLED`: Never runs the annotate command after test steps
+
+___
+
+### credentials
+
+• **credentials**: `string` - Default: `envCredentials`
+
+We need to provide this credentials for the http client. Per Default we use the credentials from the user environment.
```
Comment (adi-wan-askui) [inline docs/docs/api/06-Configuration/askui-ui-controller-client.md:59]: Could you document the environment variables and how this can be set using an object? `envCredentials` is an implementation detail and not the default. It should not be documented :laugh

File: docs/docs/api/06-Configuration/askui-ui-controller.md [hunk 3]
```diff
@@ -8,14 +8,19 @@ custom_edit_url: null
...
-- [display](#display)
-- [host](#host)
-- [logFilePath](#logfilepath)
-- [logLevel](#loglevel)
-- [minimize](#minimize)
-- [overWriteBinary](#overwritebinary)
-- [port](#port)
+ [askui UI Controller](#askui-ui-controller)
+  - [Table of Contents](#table-of-contents)
+    - [Properties](#properties)
```
Comment (adi-wan-askui) [inline docs/docs/api/06-Configuration/askui-ui-controller.md:13]: Like already commented, I don't like it as it seems redundant but maybe you have good reason. Can you explain it to me?

File: packages/askui-nodejs/src/execution/client-interface.ts [hunk 4]
```diff
@@ -2,9 +2,14 @@ import { CredentialArgs } from '../utils/http/credentials';
...
 
 export interface ClientArgs {
+  // The adress of the askui UI Controller server.
   readonly controlServerUrl?: string,
+  // Address of the askui Inference server.
   readonly controlYourUiApi?: string,
+  /* Usage of annotate command after execution of test steps.
+  You have three options: `DISABLED`, `ON_FAILURE`, `ALL`. */
   readonly annotationLevel?: AnnotationLevel,
+  // Credentials for http client.
```
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/client-interface.ts:12]: I think you can and should be more specific as this only referes to the `controlYourUiApi` (⟶ should be renamed).

File: packages/askui-nodejs/src/execution/client-interface.ts [hunk 5]
```diff
@@ -2,9 +2,14 @@ import { CredentialArgs } from '../utils/http/credentials';
 import { AnnotationLevel } from './annotation-level';
 
 export interface ClientArgs {
+  // The adress of the askui UI Controller server.
```
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/client-interface.ts:5]: I think we should use javadoc comment style for all comments for official documentation as most generators that use these comments expect that as far as I know, e.g.,

```suggestion
  /**
    * The address of the askui UI Controller server.
    */
```

**Review by adi-wan-askui — CHANGES_REQUESTED** [review]

File: docs/docs/api/06-Configuration/askui-ui-controller.md (hunk 3 above)
Comment (thinh-askui) [reply docs/docs/api/06-Configuration/askui-ui-controller.md:13]: Actually I looked at @mlikasam-askui PR 1034 and his changes to avoid conflicts. Saw his change and added it . Also used the same schema for the client.  Think it was fine as overview on the first look but the value of the three links are probably not so high

File: docs/docs/api/06-Configuration/askui-ui-controller.md [hunk 6]
```diff
@@ -68,16 +66,26 @@ The host the askui UI Controller is running on.
...
+
+On Linux use this command:
+```shell
+export LOG_LEVEL=verbose
+```
+
+For Windows you need this command:
+ ```shell
+$env:LOG_LEVEL="verbose"
+```
```
Comment (mlikasam-askui) [inline docs/docs/api/06-Configuration/askui-ui-controller.md:88]: the env LOG_LEVEL is for the `UiControlClient` and not the `UiController` itself

File: docs/docs/api/06-Configuration/askui-client.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
+---
+custom_edit_url: null
+---
+# askui Client
```
Comment (mlikasam-askui) [inline docs/docs/api/06-Configuration/askui-client.md:4]: the client is now called AskuiClient is now called UiControlCLient.
should this be renamed to askui UI Control Client?

File: docs/docs/api/06-Configuration/askui-client.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  - [Set Log Level](#set-log-level)  
+  - [Example](#example)  
+## Properties
+
+### uiControllerUrl
+
+• **uiControllerUrl**: `string` - Default: `http://localhost:6769`
+
+This provides the url where the askui UI Controller runs. The Client is also connecting
+to this address.
```
Comment (mlikasam-askui) [inline docs/docs/api/06-Configuration/askui-client.md:22]: I didn't get this part **The Client is also connecting to this address.**

File: docs/docs/api/06-Configuration/askui-client.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+The address of the askui Inference server.
+___
+
+### annotationLevel
+
+• **annotationLevel**: `AnnotationLevel` - Default: `AnnotationLevel.DISABLED`
+
+AnnotationLevel is implemented as an enum. You have three options: `DISABLED`, `ON_FAILURE`, `ALL`.
+
+`ALL`:  We run the [annotate](../../general/05-Tooling/annotate-image.md) command after the execution of each test step.
```
Comment (mlikasam-askui) [inline docs/docs/api/06-Configuration/askui-client.md:38]: i thing this should be similar to the other descriptions. We run --> Runs

File: docs/docs/api/06-Configuration/askui-client.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+For Windows you need this command:
+ ```shell
+$env:LOG_LEVEL="verbose"
+```
+## Example
+
+```typescript
+ let aui = new UiControlClient({
+    askuiUiControllerServerUrl: 'http://localhost:6769',
+    inferenceServerUrl: 'https://inference.askui.com',
```
Comment (mlikasam-askui) [inline docs/docs/api/06-Configuration/askui-client.md:74]: these are the default values, I think we can remove them from this example.

File: packages/askui-nodejs/src/main.ts [hunk 11]
```diff
@@ -1,3 +1,3 @@
 export { UiController } from './lib';
-export { UiControlClient } from './execution';
+export { UiControlClient, AnnotationLevel } from './execution';
```
Comment (mlikasam-askui) [inline packages/askui-nodejs/src/main.ts:2]: Nice Idea :+1:

File: packages/askui-nodejs/src/lib/ui-controller-args.ts [hunk 12]
```diff
@@ -1,6 +1,24 @@
...
+ * @param port The port the askui UI Controller is running on.
+ * @param host The host the askui UI Controller is running on.
+ * @param minimize askui UI Controller will be started as minimized window.
+ * @param overWriteBinary Download the provided Version of askui UI Controller.
+ * If a version is already downloaded. This version will be overwritten
+ * @param logLevel You can use different types of Log Level.
+ * Options: "fatal", "error", "warn", "info", "debug", "trace", "silent", "verbose".
+ * @param logFilePath It is possible to specify a path for your log files.
+ * Per default we create the askui-server.log file and askui folder in your temp folder.
+ */
```
Comment (mlikasam-askui) [inline packages/askui-nodejs/src/lib/ui-controller-args.ts:18]: I though the incode description needed to be Done with [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) standard. Or at least this how we did it in the file `askui/packages/askui-nodejs/src/utils/transformations.ts`

File: docs/docs/api/06-Configuration/askui-ui-controller.md [hunk 13]
```diff
@@ -68,16 +57,16 @@ The host the askui UI Controller is running on.
...
-It is possible to specify a path for your log files. Logs are written to the stdout per default.
+It is possible to specify a path for your log files. Per default we create the askui-server.log file and askui folder in your temp folder.
 ___
 
 ### logLevel
 
 • **logLevel**: `LogLevels` - Default: `debug`
 
-You can use different types of Log Level. We provide this options: `"fatal"`, `"error"`, `"warn"`, `"info"`, `"debug"`, `"trace"`, `"silent"`, `"verbose"`. All loglevel are defined in an `LogLevels` enum.
+You can use different types of Log Level. We provide this options: `"fatal"`, `"error"`, `"warn"`, `"info"`, `"debug"`, `"trace"`, `"silent"`, `"verbose"`. All log levels are defined in an `LogLevels` enum. You can export all log levels.
```
[the remark this answers is not in the document]
Comment (thinh-askui) [reply docs/docs/api/06-Configuration/askui-ui-controller.md:69]: Can be removed. Was from the older version with setting environment variable

File: docs/docs/api/06-Configuration/askui-client.md (hunk 8 above)
Comment (thinh-askui) [reply docs/docs/api/06-Configuration/askui-client.md:22]: the ui control client is also connecting to the url of the ask UI Controller? or did i misunderstand something

**Review by mlikasam-askui — APPROVED** [review]

File: docs/docs/api/06-Configuration/askui-ui-control-client.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+  - [Set Log Level](#set-log-level)  
+  - [Example](#example)  
+## Properties
+
+### uiControllerUrl
+
+• **uiControllerUrl**: `string` - Default: `http://localhost:6769`
+
+This provides the url where the askui UI Controller runs. The askui UI Control Client is also connecting
+to this address.
```
Comment (adi-wan-askui) [inline docs/docs/api/06-Configuration/askui-ui-control-client.md:22]: Seems like we can shorten this to `The url of the askui UI Controller controlling the OS.` What do you think?

File: docs/docs/api/06-Configuration/askui-ui-control-client.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+
+`ON_FAILURE`: Runs the annotate command if the test step fails 
+
+
+`DISABLED`: Never runs the annotate command after test steps
+___
+
+### credentials
+
+• **credentials**: `string` - Default: `environment-tenant`, `environment-token`, `environment-email`
```
Comment (adi-wan-askui) [inline docs/docs/api/06-Configuration/askui-ui-control-client.md:49]: There is not default value for `credentials` as far as I know. It is optional.

Also it is no `string` from my perspective but rather an `object`. I would document the name of the properties of that object and their types. Maybe, we can do this like jsdoc does it, e.g., esee https://jsdoc.app/tags-type.html#:~:text=An%20object%20called%20'myObj'%20with,c'%20(any%20type).&text=JSDoc%20supports%20Closure%20Compiler's%20syntax,is%20contained%20in%20the%20array.

File: docs/docs/api/06-Configuration/askui-ui-control-client.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+
+We need to provide credentials for the authentication of the askui Inferrence Server. If you want to set your own credentials you need to provide the following arguments `tenant`, `token`, `email`. Per default we read the environment variables.
```
Comment (adi-wan-askui) [inline docs/docs/api/06-Configuration/askui-ui-control-client.md:51]: It would be good if we documented the names of the environment variables and how they map to the properties of the credentials object.

File: docs/docs/api/06-Configuration/askui-ui-control-client.md [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+___
+
+## Set Log Level
+
+Starting the askui library the askui UI Controller will write [logs](../06-Configuration/askui-ui-controller.md#loglevel) in a default/passed path. ([logfile path](../06-Configuration/askui-ui-controller.md#logfilepath)).
+Only the askui Client can write logs to the stdout while using the askui library.
+You can set all log levels(`"fatal"`, `"error"`, `"warn"`, `"info"`, `"debug"`, `"trace"`, `"silent"`, `"verbose"`) for the client.
```
Comment (adi-wan-askui) [inline docs/docs/api/06-Configuration/askui-ui-control-client.md:58]: Does this really also set the logs of the askui UI Controller? I thought that this was only for the client lib. 

Maybe, we can improve the wording a bit. What about the following?

You can set the log level using the `LOG_LEVEL` environment variable. The following values are available:
- `"fatal"`, 
- `"error"`, 
- `"warn"`, 
- `"info"`, 
- `"debug"`, 
- `"trace"`, 
- `"silent"` and 
- `"verbose"`.

Maybe, we can also add an explanation of some log levels as at least I would not be sure of what `"silent"` actually means.

File: docs/docs/api/06-Configuration/askui-ui-controller.md [hunk 18]
```diff
@@ -105,15 +94,15 @@ The port the askui UI Controller is running on.
...
 
 describe('jest with askui', () => {
   
   // Server for controlling the operating system
-  let askuiControlServer: AskuiControlServer;
+  let uiController: UiController;
   
   beforeAll(async () => {
-    askuiControlServer = new AskuiControlServer({
+    uiController = new UiController()({
```
Comment (adi-wan-askui) [inline docs/docs/api/06-Configuration/askui-ui-controller.md:105]: rm `()`

File: packages/askui-nodejs/src/execution/ui-controller-client-interface.ts [hunk 19]
```diff
@@ -1,6 +1,17 @@
 import { CredentialArgs } from '../utils/http/credentials';
 import { AnnotationLevel } from './annotation-level';
 
+/**
+ * Provide configuration options for the askui UI Control Client
```
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-controller-client-interface.ts:5]: I would rm `Provide` and uppercase `Configuration` as we are describing something static

File: packages/askui-nodejs/src/execution/ui-controller-client-interface.ts [hunk 20]
```diff
@@ -1,6 +1,17 @@
...
+ *
+ * @param {string} uiControllerUrl - The adress of the askui UI Controller server.
+ * @param {string} inferenceClientUrl - Address of the askui Inference server.
+ * @param {AnnotationLevel} annotationLevel - Usage of annotate command
+ * after execution of test steps.
+ * You have three options: `DISABLED`, `ON_FAILURE`, `ALL`.
+ * @param {CredentialArgs} credentials - We need to provide credentials for
+ * the authentication of the askui Inferrence Server.
```
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-controller-client-interface.ts:13]: typo in `Inference`

File: packages/askui-nodejs/src/lib/ui-controller-args.ts [hunk 21]
```diff
@@ -1,6 +1,26 @@
...
+ * @param {string} binaryVersion - Choose the version of the askui UI Controller Server.
+ * @param {number} port - The port the askui UI Controller is running on.
+ * @param {string} host - The host the askui UI Controller is running on.
+ * @param {boolean} minimize - askui UI Controller will be started as minimized window.
+ * @param {boolean} overWriteBinary - Download the provided Version of askui UI Controller.
+ * If a version is already downloaded. This version will be overwritten
+ * @param {LogLevels} logLevel - You can use different types of Log Level.
+ * Options: "fatal", "error", "warn", "info", "debug", "trace", "silent", "verbose".
+ * @param {string} logFilePath - It is possible to specify a path for your log files.
+ * Per default we create the askui-server.log file and askui folder in your temp folder.
```
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/lib/ui-controller-args.ts:19]: Can you specify the defaults in the in-code docs for the different options?

File: docs/docs/api/06-Configuration/askui-ui-control-client.md (hunk 17 above)
Comment (mlikasam-askui) [reply docs/docs/api/06-Configuration/askui-ui-control-client.md:58]: This is not the environment variable, during the starting of the ui-controller we can select the log_levwel there, default is set to verbose as far as I remember.

**Review by mlikasam-askui — APPROVED** [review]

**Review by adi-wan-askui — APPROVED** [review]
